testsuite/forge/src/backend/k8s/swarm.rs

```diff
@@ -280,3 +298,100 @@ fn load_root_key(root_key_bytes: &[u8]) -> Ed25519PrivateKey {
 fn load_tc_key(tc_key_bytes: &[u8]) -> Ed25519PrivateKey {
     Ed25519PrivateKey::try_from(tc_key_bytes).unwrap()
 }
+
+fn clean_k8s_cluster_internal() {
+    clean_k8s_cluster(DEFL_HELM_REPO_NAME.to_string(), DEFL_NUM_VALIDATORS)
+        .map_err(|err| format_err!("Failed to clean k8s cluster with new genesis: {}", err))
+        .unwrap()
+}
+
+pub fn clean_k8s_cluster(
+    helm_repo: String,
+    base_num_validators: usize,
+) -> Result<(), anyhow::Error> {
+    let rt = Runtime::new().unwrap();
+
+    // get the previous chain era
+    let raw_helm_values = Command::new(HELM_BIN)
+        .arg("get")
+        .arg("values")
+        .arg("diem")
+        .arg("--output")
+        .arg("json")
+        .output()
+        .unwrap();
+
+    // parse genesis
+    let helm_values = String::from_utf8(raw_helm_values.stdout).unwrap();
+    let v: Value = serde_json::from_str(&helm_values).unwrap();
+    let era = &v["genesis"]["era"];
+    let num_validators = &v["genesis"]["numValidators"];
+    let new_era = if era == 1 { 2 } else { 1 };
+
+    println!("genesis.era: {} --> {}", era, new_era);
+    println!(
+        "genesis.numValidators: {} --> {}",
+        num_validators, base_num_validators
+    );
+
+    // upgrade testnet
+    let testnet_upgrade_args = [
+        "upgrade",
+        "diem",
+        &format!("{}/testnet", helm_repo),
+        "--reuse-values",
+        "--set",
+        &format!("genesis.era={}", new_era),
+        "--set",
+        &format!("genesis.numValidators={}", base_num_validators),
+    ];
+    println!("{:?}", testnet_upgrade_args);
+    let testnet_upgrade_output = Command::new(HELM_BIN)
+        .stdout(Stdio::inherit())
+        .args(&testnet_upgrade_args)
+        .output()
+        .expect("failed to helm upgrade diem");
+    assert!(testnet_upgrade_output.status.success());
+
+    // upgrade validators in parallel
+    (0..base_num_validators).into_par_iter().for_each(|i| {
+        let validator_upgrade_args = [
+            "upgrade",
+            &format!("val{}", i),
+            &format!("{}/diem-validator", helm_repo),
+            "--reuse-values",
+            "--set",
+            &format!("chain.era={}", new_era),
+        ];
+        println!("{:?}", validator_upgrade_args);
+        let validator_upgrade_output = Command::new(HELM_BIN)
+            .stdout(Stdio::inherit())
+            .args(&validator_upgrade_args)
+            .output()
+            .expect("failed to helm upgrade diem");
+        assert!(validator_upgrade_output.status.success());
+    });
+
+    let kube_client = rt.block_on(K8sClient::try_default()).unwrap();
+
+    rt.block_on(async {
+        diem_retrier::retry_async(k8s_retry_strategy(), || {
+            let jobs: Api<Job> = Api::namespaced(kube_client.clone(), "default");
+            Box::pin(async move {
+                let job_name = format!("diem-testnet-genesis-e{}", new_era);
```

Comment:
shall we add a sanity check here? something like fetch all the genesis jobs and validating totally name liked job numbers etc before we are looping checking the job status?

Reply:
I think this loop handles that? We're only checking the creation and running (to completion) of the new genesis job.  Ideally there's only one genesis job at once, but if there are multiple, one of them will fail, causing Forge test to fail, in which case an operator can come remediate manually.

Reply:
ok, I c this is for testnet genesis job. Another question, do we need to check validator upgrading with new era above? I did see sometimes the same validator has multi pods with different era numbers

Reply:
>  multi pods with different era numbers

Still trying to figure this out. I think it's an artifact of using helm (or the TF helm provider) with such a large number of helm releases. I'll run a few tests with this new method and see if I can repro that issue.

Reply:
yea, I think we can postpone this point until it reproduce again since we have exist solution to fix

Reply:
Yea I think the issue is only with the TF provider. I'm able to reset my cluster with `forge` many times without issue.